plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/type/TestTypeInfoUtils.java

```diff
@@ -0,0 +1,132 @@
+        testNumeric("NUMERIC");
+    }
+
+    @Test
+    public void testBignumeric()
+    {
+        testNumeric("BIGNUMERIC");
+    }
+
+    private void testNumeric(String typeName)
+    {
+        assertTypeInfo(typeName, "NUMERIC(10, 0)").isEqualTo(getDecimalTypeInfo(10, 0));
+        assertTypeInfo(typeName + "(1)", "NUMERIC(1, 0)").isEqualTo(getDecimalTypeInfo(1, 0));
+        assertTypeInfo(typeName + "(5)", "NUMERIC(5, 0)").isEqualTo(getDecimalTypeInfo(5, 0));
+        assertTypeInfo(typeName + "(38)", "NUMERIC(38, 0)").isEqualTo(getDecimalTypeInfo(38, 0));
+        assertTypeInfo(typeName + "(1, 1)", "NUMERIC(1, 1)").isEqualTo(getDecimalTypeInfo(1, 1));
+        assertTypeInfo(typeName + "(10, 5)", "NUMERIC(10, 5)").isEqualTo(getDecimalTypeInfo(10, 5));
+        assertTypeInfo(typeName + "(38, 38)", "NUMERIC(38, 38)").isEqualTo(getDecimalTypeInfo(38, 38));
+
+        assertThatThrownBy(() -> getTypeInfoFromTypeString(typeName + "(0)"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("invalid decimal precision: 0");
+
+        assertThatThrownBy(() -> getTypeInfoFromTypeString(typeName + "(39)"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("invalid decimal precision: 39");
+
+        assertThatThrownBy(() -> getTypeInfoFromTypeString(typeName + "(38,39)"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("invalid decimal scale: 39");
+
+        assertThatThrownBy(() -> getTypeInfoFromTypeString(typeName + "(4,5)"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("decimal precision (4) is greater than scale (5)");
+    }
+
+    @Test
+    public void testArray()
+    {
+        assertTypeInfo("ARRAY<INT64>")
+                .isEqualTo(getListTypeInfo(getPrimitiveTypeInfo("INT64")));
+    }
+
+    @Test
+    public void testStruct()
+    {
+        assertTypeInfo("STRUCT<x INT64, y STRING>")
+                .isEqualTo(getStructTypeInfo(
+                        List.of("x", "y"),
+                        List.of(getPrimitiveTypeInfo("INT64"),
+                                getPrimitiveTypeInfo("STRING"))));
+
+        assertTypeInfo("STRUCT<x INT64, y ARRAY<STRING>>")
+                .isEqualTo(getStructTypeInfo(
+                        List.of("x", "y"),
+                        List.of(getPrimitiveTypeInfo("INT64"),
+                                getListTypeInfo(getPrimitiveTypeInfo("STRING")))));
+    }
+
+    private static ObjectAssert<TypeInfo> assertTypeInfo(String typeString)
+    {
+        assertThat(getBigQueryTypeInfo(typeString))
+                .hasToString(typeString);
+
+        return assertThat(getTypeInfoFromTypeString(typeString))
+                .hasToString(typeString);
+    }
+
+    private static ObjectAssert<TypeInfo> assertTypeInfo(String typeString, String toString)
+    {
+        assertThat(getBigQueryTypeInfo(typeString))
```

Comment:
this is testing the same thing twice:

getBigQueryTypeInfo(typeString) is TypeInfoUtils.getTypeInfoFromTypeString(typeString)